woocommerce/woocommerce
# Fix broken coming soon template in dataview

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/49699

This PR fixes the broken "coming soon" template in the dataview. The issue is that the preview iframe [height is set dynamically based on the content height](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-preview/auto.js#L101), while our pattern height is set to 100vh, causing dependency issues. To fix this, we could set the content height to a fixed value, but this isn't ideal because the content height varies. Instead, the best solution is to set the aspect ratio of the content body to 1/1, making the height and width equal, since the preview block is a square. This allows the preview iframe to calculate the height correctly. However, if Gutenberg changes the preview block to a rectangle in the future, this solution may need updating. But for now, it's the best approach IMO.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site using this branch.
2. Skip Core Profiler
3. Go to `/wp-admin/site-editor.php?postType=wp_template`
4. Search for the `Coming Soon` template
5. Observe the preview should be displayed correctly
6. Click on the template and do some customizations
7. Observe the preview should be still displayed correctly

![Screenshot 2024-07-22 at 15 21 09](https://github.com/user-attachments/assets/ade48a85-dc0c-448d-b1b4-31a0b33eae13)
![Screenshot 2024-07-22 at 15 21 44](https://github.com/user-attachments/assets/6efa3bf6-e067-4779-a6bd-c335100502b6)




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
